lxd/operations/operations.go

```diff
@@ -123,7 +124,7 @@ type Operation struct {
 
 // OperationCreate creates a new operation and returns it. If it cannot be
 // created, it returns an error.
-func OperationCreate(s *state.State, projectName string, opClass OperationClass, opType db.OperationType, opResources map[string][]string, opMetadata interface{}, onRun func(*Operation) error, onCancel func(*Operation) error, onConnect func(*Operation, *http.Request, http.ResponseWriter) error) (*Operation, error) {
+func OperationCreate(s *state.State, projectName string, opClass OperationClass, opType db.OperationType, opResources map[string][]string, opMetadata interface{}, onRun func(*Operation) error, onCancel func(*Operation) error, onConnect func(*Operation, *http.Request, http.ResponseWriter) error, r *http.Request) (*Operation, error) {
```

Comment:
I think I would have gone with just passing r.Context() as an argument to this function to avoid tightly coupling the requester functionality to the `http` package. That way it would be easier to create a context with a username and protocol values manually from other sources if ever needed.

Reply:
The main reasons why I didn't is that we don't have any such cases currently and passing a context without treating it like a context (not supporting cancelation/timeout through it) felt weird. I also thought we may find it useful to have access to the rest of the request fields in there (though so far haven't made use of it).

Reply:
Yes that makes sense thanks for clarifying :)